modules/kargoBidAdapter.js

```diff
@@ -27,58 +41,103 @@ export const spec = {
+        ]
       },
-      prebidRawBidRequests: validBidRequests
-    }, spec._getAllMetadata(bidderRequest, tdid));
+      imp: impressions,
+      user: spec._getUserIds(tdid, bidderRequest.uspConsent, bidderRequest.gdprConsent),
+      eids: firstBidRequest.userIdAsEids
+    });
 
-    // User Agent Client Hints / SUA
-    const uaClientHints = deepAccess(firstBidRequest, 'ortb2.device.sua');
-    if (uaClientHints) {
-      transformedParams.device.sua = pick(uaClientHints, ['browsers', 'platform', 'mobile', 'model']);
+    const reqCount = spec._getRequestCount()
+    if (reqCount != null && reqCount > 0) {
+      krakenParams.requestCount = reqCount
+    }
+
+    if (currency != null && currency != US_DOLLAR_KEY) {
+      krakenParams.cur = currency
     }
 
     // Pull Social Canvas segments and embed URL
-    const socialCanvas = deepAccess(firstBidRequest, 'params.socialCanvas');
-    if (socialCanvas) {
-      transformedParams.socialCanvasSegments = socialCanvas.segments;
-      transformedParams.socialEmbedURL = socialCanvas.embedURL;
+    const socialCanvas = deepAccess(firstBidRequest, SOCIAL_CANVAS_BID_KEY);
+
+    if (socialCanvas != null) {
+      krakenParams.socan = socialCanvas
+    }
+
+    // User Agent Client Hints / SUA
+    const uaClientHints = deepAccess(firstBidRequest, SUA_BID_KEY);
+    if (uaClientHints) {
+      const suaValidAttributes = []
+
+      for (const suaKey of SUA_ATTRIBUTES) {
+        const suaValue = uaClientHints[suaKey]
+        if(!suaValue) {
+          continue
+        }
+
+        // Do not pass any empty strings
+        if(typeof suaValue == 'string' && suaValue.trim() === '') {
+          continue
+        }
+
+        switch (suaKey) {
+          case MOBILE, SOURCE:
+            if (suaValue < 1) {
+              return;
+            };
+          default:
+            suaValidAttributes.push(suaKey);
+        }
+      }
+
+      krakenParams.device.sua = pick(uaClientHints, suaValidAttributes);
     }
 
-    const encodedParams = encodeURIComponent(JSON.stringify(transformedParams));
+    const validPageId = spec._getLocalStorageSafely(PAGE_VIEW_ID_LOCAL_KEY) != null 
+    const validPageTimestamp = spec._getLocalStorageSafely(PAGE_VIEW_TIMESTAMP_LOCAL_KEY) != null
+    const validPageUrl = spec._getLocalStorageSafely(PAGE_VIEW_URL_LOCAL_KEY) != null
+ 
+    const page = {}
+    if(validPageId) {
+      page.id = spec._getLocalStorageSafely(PAGE_VIEW_ID_LOCAL_KEY)
+    }
+    if(validPageTimestamp) {
+      page.timestamp = Number(spec._getLocalStorageSafely(PAGE_VIEW_TIMESTAMP_LOCAL_KEY))
+    }
+    if(validPageUrl) {
+      page.url = spec._getLocalStorageSafely(PAGE_VIEW_URL_LOCAL_KEY)
+    }
+    if (!_.isEmpty(page)) {
+      krakenParams.page = page
+    }
+    
     return Object.assign({}, bidderRequest, {
-      method: 'GET',
-      url: `${HOST}/api/v2/bid`,
-      data: `json=${encodedParams}`,
+      method: 'POST',
+      url: `${HOST}/api/v1/prebid`,
```

Comment:
Nit: probably not necessary but should the entire route just be a const?

Reply:
In the latest changes I've added a "HOST" and "REQUEST_ENDPOINT" consts as part of a BIDDER object. That would allow for future reusability of both individually, should the need arise.